### PR TITLE
Reproduce and fix a query of death

### PR DESCRIPTION
**Pull request by bdarnell**
I'm not sure if this is the right fix. The `default` case of this switch panics if the value is nil, so I added a check for nil, but instead we may want to turn that nil into a `DNull` somewhere else. I haven't dug around to see where exactly the nil is coming from. 

The way in which the newly-added test failed was interesting too: the executor panics, and the panic is caught by the HTTP server, so the process doesn't die. Instead, the test times out, due to a leak in the stopper. The second commit in this branch fixes that.

### CODE REVIEW

File: sql/testdata/select_non_covering_index [hunk 1]
```diff
@@ -63,3 +63,25 @@ EXPLAIN SELECT * FROM t WHERE c > 0 ORDER BY c
...
 1 scan       t@c /1-
 1 scan       t@primary
+
+statement ok
+INSERT INTO t VALUES (7, 8, NULL)
+
+query I
+SELECT c FROM t WHERE a=7
+----
+NULL
```
Comment (petermattis) [inline sql/testdata/select_non_covering_index]: Seems like there is another bug somewhere. Why is selecting a `NULL` value causing a crash?  This also seems unrelated to non-covering indexes as you're using the primary index here.

Comment (bdarnell) [inline sql/testdata/select_non_covering_index]: Ah, you're right. It's not a non-covering index (although it is a column that is stored in the body rather than making up part of the key). So the test is miscategorized. The panic happens whether the secondary indexes are there or not (as demonstrated by the `boolean_table` test below, which I meant to remove before submitting this PR).

Comment (petermattis) [inline sql/testdata/select_non_covering_index]: I think the change to `sql.Executor` is good defense, but there is also a bug in `scanNode.filterRow`. The loop over the visible columns should not break when a non-nullable column is found. This was the root cause of a `nil` datum being returned to the `Executor`. I'll send out a PR shortly. I think this change is still worthwhile, just move the tests to `testdata/select`.

**Comment by petermattis** [comment]
LGTM modulo moving the tests to `testdata/select`.

**Comment by bdarnell** [comment]
If we never expect nil values to make it this far, then adding a `case nil:` will just mask errors in the future. After your PR we no longer have a way to test whether nils are handled gracefully (and you've already put the tests in `testdata/select`), so I'm going to remove the SQL parts of this change and just merge the changes to Stopper.
